pfcpiface/conn.go

```diff
@@ -136,35 +160,52 @@ func pfcpifaceMainLoop(upf *upf, accessIP, coreIP, sourceIP, smfName string) {
 				}
 			}
 		case message.MsgTypeAssociationSetupResponse:
-			cpConnected = handleAssociationSetupResponse(msg, addr, sourceIP, accessIP)
+			cpConnected = handleAssociationSetupResponse(msg, pPkt.Address, sourceIP, accessIP)
 			if manageConnection {
 				// pass on information to go routine that result of association response
 				cpConnectionStatus <- cpConnected
 			}
 		case message.MsgTypePFDManagementRequest:
-			outgoingMessage = pconn.handlePFDMgmtRequest(upf, msg, addr, sourceIP)
+			outgoingMessage = pconn.handlePFDMgmtRequest(upf, msg, pPkt.Address, sourceIP)
 		case message.MsgTypeSessionEstablishmentRequest:
-			outgoingMessage = pconn.handleSessionEstablishmentRequest(upf, msg, addr, sourceIP)
+			outgoingMessage = pconn.handleSessionEstablishmentRequest(upf, msg, pPkt.Address, sourceIP)
 		case message.MsgTypeSessionModificationRequest:
-			outgoingMessage = pconn.handleSessionModificationRequest(upf, msg, addr, sourceIP)
+			outgoingMessage = pconn.handleSessionModificationRequest(upf, msg, pPkt.Address, sourceIP)
 		case message.MsgTypeHeartbeatRequest:
-			outgoingMessage = handleHeartbeatRequest(msg, addr)
+			outgoingMessage = handleHeartbeatRequest(msg, pPkt.Address)
 		case message.MsgTypeSessionDeletionRequest:
-			outgoingMessage = pconn.handleSessionDeletionRequest(upf, msg, addr, sourceIP)
+			outgoingMessage = pconn.handleSessionDeletionRequest(upf, msg, pPkt.Address, sourceIP)
 		case message.MsgTypeAssociationReleaseRequest:
-			outgoingMessage = handleAssociationReleaseRequest(msg, addr, sourceIP, accessIP)
+			outgoingMessage = handleAssociationReleaseRequest(msg, pPkt.Address, sourceIP, accessIP)
 			cleanupSessions()
 		default:
 			log.Println("Message type: ", msg.MessageTypeName(), " is currently not supported")
-			continue
+			return
 		}
 
 		// send the response out
 		if outgoingMessage != nil {
-			if _, err := conn.WriteTo(outgoingMessage, addr); err != nil {
+			if _, err := conn.WriteTo(outgoingMessage, pPkt.Address); err != nil {
 				log.Fatalln("Unable to transmit association setup response", err)
 			}
 		}
 
 	}
+
+	go pfcpPacketReader(pfcpRcvdPktsChan)
```

Comment:
make sure we dont leak goroutines. explicit exit criteria should be there. usually the pattern is to have a bool chan from main -> goroutine to send termination message and goroutine -> main to send ack.